pwiz_tools/Skyline/Util/Util.cs

```diff
@@ -1957,6 +1966,36 @@ public static string NullableDoubleToString(double? d)
     /// </summary>
     public static class Assume
     {
+
+        private static string _machineNamePatternForInvokeDebuggerOnFail; // When set, if machine name contains this string we will invoke the debugger rather than fail.
+        public class DebugOnFail : IDisposable
+        {
+            private string _initialValue;
+
+            public DebugOnFail(string machineNamePattern)

```

Comment:
Let's make this a Func<bool> instead of a machine name. That way the function could be anything and it could be testing for a set of 5 different machines or whatever. You could put utility functions in our test utility code without needing to implement them inside Assume.